.azure/pipelines/jobs/default-build.yml

```diff
@@ -300,6 +295,15 @@ jobs:
           artifactType: Container
           parallel: true
 
+    - ${{ if eq(parameters.agentOs, 'Windows') }}:
+      - powershell: $(Build.SourcesDirectory)/eng/scripts/UploadCores.ps1 -ProcDumpOutputPath artifacts/dumps/
+        condition: failed()
+        displayName: Upload cores
+    - ${{ if ne(parameters.agentOs, 'Windows') }}:
+      - script: $(Build.SourcesDirectory)/eng/scripts/upload-cores.sh
+        condition: failed()
+        displayName: Upload cores
+
```

Comment:
@javiercn these scripts handle the hang dumps too. In your previous setup, those dumps were uploaded _first_. I mimicked this in 0d0103b4c047, uploading all cores before any requested `parameters.artifacts`.

In this, I moved all core uploads _after_ `parameters.artifacts` uploads because binary logs et cetera are more likely to provide useful information when builds fail.

Does this make sense to you❔